src/Emoji.ts

```diff
@@ -0,0 +1,24 @@
+// This function returns whether the first code point of the provided string is
+// within a range that contains emoji. Some of the codepoints in these ranges
+// are unassigned so this test is an approximation to the real thing.
+// Ranges taken from: https://stackoverflow.com/questions/30757193/
+// For a complete list of all assigned codepoints, go to:
+//   https://unicode.org/Public/emoji/11.0/emoji-data.txt
+export function isEmoji(str: string): boolean {
```

Comment:
@jerch thoughts? Looks very similar to what CharWidth.ts is doing

Reply:
Yupp thats pretty similar. The short circuit for lower chars is important imho, otherwise it will run 10 checks for every single char. The conditions could even be further tweaked to a binary search, which would lower the needed checks to ~3 per char. But since input is nothing coming in MB/s this is good as it is.
Btw `codePointAt` cannot be used since its not supported by IE11. You need a polyfill that converts 2 UTF16 surrogates to a UTF32 codepoint. something like this: https://github.com/xtermjs/xterm.js/blob/d7e5977d9ab825c5c9941f7a81371b6df897c69c/src/core/input/TextDecoder.ts#L28
Once we have multiple unicode version support we should move it to the v11 definitions.

Reply:
> Btw codePointAt cannot be used since its not supported by IE11.

I think we'll probably drop IE support soon, you could check if the function exists and use it, otherwise return `false` (for IE).